From ravibabu@ti.com  Fri Jul 10 15:19:23 2009
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Fri,  3 Jul 2009 13:18:45 +0530
Subject: USB: otg: fix module reinsert issue
To: linux-usb@vger.kernel.org
Cc: david-b@pacbell.net, felipe.balbi@nokia.com, Ajay Kumar Gupta <ajay.gupta@ti.com>, "Ravi, Babu" <ravibabu@ti.com>
Message-ID: <1246607325-22927-1-git-send-email-ravibabu@ti.com>

From: Ajay Kumar Gupta <ajay.gupta@ti.com>

Platform_device instance (pd) is not set to NULL in
usb_nop_xceiv_unregister() causing usb_nop_xceiv_register()
to fail during module reinsert.

From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Signed-off-by: Babu Ravi <ravibabu@ti.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>

---
 drivers/usb/otg/nop-usb-xceiv.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/otg/nop-usb-xceiv.c
+++ b/drivers/usb/otg/nop-usb-xceiv.c
@@ -53,6 +53,7 @@ EXPORT_SYMBOL(usb_nop_xceiv_register);
 void usb_nop_xceiv_unregister(void)
 {
 	platform_device_unregister(pd);
+	pd = NULL;
 }
 EXPORT_SYMBOL(usb_nop_xceiv_unregister);
 
